packages/xo-web/src/xo-app/hub/resource-form.js

```diff
@@ -3,38 +3,65 @@ import _ from 'intl'
 import decorate from 'apply-decorators'
 import Icon from 'icon'
 import React from 'react'
+import SingleLineRow from 'single-line-row'
 import Tooltip from 'tooltip'
-import { Container } from 'grid'
-import { SelectPool } from 'select-objects'
-import { error } from 'notification'
+import { Container, Col } from 'grid'
+import { find, first } from 'lodash'
+import { SelectPool, SelectSr } from 'select-objects'
+import { isSrWritable } from 'xo'
 import { injectState, provideState } from 'reaclette'
 
 export default decorate([
   provideState({
     initialState: ({ multi }) => ({
       pools: multi ? [] : undefined,
+      mapPoolsSrs: {},
     }),
     effects: {
       onChangePool(__, pools) {
-        const noDefaultSr = Array.isArray(pools)
-          ? pools.some(pool => pool.default_SR === undefined)
-          : pools.default_SR === undefined
-        if (noDefaultSr) {
-          error(_('hubSrErrorTitle'), _('noDefaultSr'))
-        } else {
-          this.props.onChange({
-            pools,
-            pool: pools,
-          })
-          return {
-            pools,
+        this.props.onChange({
+          pools,
+          mapPoolsSrs: this.state.mapPoolsSrs,
+        })
+        const _mapPoolsSrs = {}
+        if (Array.isArray(pools)) {
+          for (const pool of pools) {
+            _mapPoolsSrs[pool.id] = pool.default_SR
           }
+        } else {
+          _mapPoolsSrs[pools.id] = pools.default_SR
+        }
+        return {
+          pools,
+          mapPoolsSrs: _mapPoolsSrs,
         }
       },
+      onChangeSr(__, sr) {
+        const { mapPoolsSrs, pools } = this.state
+        const _mapPoolsSrs = { ...mapPoolsSrs, [sr.$pool]: sr.id }
+        this.props.onChange({
```

Comment:
What's this `onChange`?

Reply:
It's the implementation of the `form/modal`. But the render component of the modal should be controlled.

Reply:
Ok got it. (We should add some documentation on the form modal :) ) Then yes, the content of the modal must be fully controlled.

Reply:
I agree :+1: